### ISSUE TITLE
False positive for Style/VariableName cop

### ISSUE THREAD

**Issue by sferik** [rubocop/rubocop#1286]
The following code causes the `Style/VariableName` cop to emit the following offense: `Use snake_case for variables.`

``` ruby
fields.select { |field| self == field.group }
```

Moving `self` to the right-hand side causes the offense not to be reported:

``` ruby
fields.select { |field| field.group == self}
```
